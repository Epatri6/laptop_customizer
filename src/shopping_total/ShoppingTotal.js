import React from 'react';
import USCurrencyFormat from '../util/USCurrencyFormat';
import './ShoppingTotal.css'

export default class ShoppingTotal extends React.Component {

  //calculates the price of selected items
  total = () => {
    const selectedFeatures = this.props.selectedFeatures;
    return Object.keys(selectedFeatures).reduce(
      (acc, curr) => acc + selectedFeatures[curr].cost,
      0
    );
  };

  render() {
    return (
      <div className="summary__total">
        <div className="summary__total__label">Total</div>
        <div className="summary__total__value">
          {USCurrencyFormat.format(this.total())}
        </div>
      </div>
    );
  }

  static defaultProps = {
    selectedFeatures: {}
  }
}
