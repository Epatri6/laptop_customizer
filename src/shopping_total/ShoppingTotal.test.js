import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingTotal from './ShoppingTotal';
import renderer from 'react-test-renderer';

describe('Testing ShoppingTotal Component', () => {

  const testProps = {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    'Operating System': {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    }
  };

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingTotal selectedFeatures={testProps}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingTotal />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<ShoppingTotal selectedFeatures={testProps}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<ShoppingTotal />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});