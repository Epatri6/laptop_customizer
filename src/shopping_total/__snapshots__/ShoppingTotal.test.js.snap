// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingTotal Component renders as expected 1`] = `
<div
  className="summary__total"
>
  <div
    className="summary__total__label"
  >
    Total
  </div>
  <div
    className="summary__total__value"
  >
    $900.00
  </div>
</div>
`;

exports[`Testing ShoppingTotal Component renders as expected without props 1`] = `
<div
  className="summary__total"
>
  <div
    className="summary__total__label"
  >
    Total
  </div>
  <div
    className="summary__total__value"
  >
    $0.00
  </div>
</div>
`;
